addons/storyquest_bootstrap/plugin.gd

```diff
@@ -0,0 +1,165 @@
+	var storyquest_uids := {
+		"intro": ResourceUID.id_to_text(ResourceUID.create_id()),
+		"outro": ResourceUID.id_to_text(ResourceUID.create_id()),
+	}
+
+	for i in template_info:
+		var directory: String = template_info[i]["components_path"]
+		error = DirAccess.make_dir_recursive_absolute(
+			storyquest_path + directory.replacen("template_", "")
+		)
+		assert(error == OK)
+
+		var scene_path: String = template_info[i]["scene_path"].replacen("template_", "")
+		var dialogue_path: String = template_info[i]["dialogue_path"].replacen("template_", "")
+
+		var template_dialogue: DialogueResource = ResourceLoader.load(
+			TEMPLATE_PATH + template_info[i]["dialogue_path"]
+		)
+
+		# Saving the raw text because ResourceSaver.save() fails with ERR_FILE_UNRECOGNIZED
+		var dialogue_file: FileAccess = FileAccess.open(
+			storyquest_path + dialogue_path, FileAccess.WRITE
+		)
+		dialogue_file.store_string(template_dialogue.raw_text)
+		dialogue_file.close()
+
+		var template_dialogue_uid := ResourceUID.id_to_text(
+			ResourceLoader.get_resource_uid(TEMPLATE_PATH + template_info[i]["dialogue_path"])
+		)
+
+		var template_scene_uid := ResourceUID.id_to_text(
+			ResourceLoader.get_resource_uid(TEMPLATE_PATH + template_info[i]["scene_path"])
+		)
+		var template_image_uid: String
+		var image_path: String
+		var template_image_path = template_info[i].get("image_path")
+		if template_image_path:
+			template_image_uid = ResourceUID.id_to_text(
+				ResourceLoader.get_resource_uid(TEMPLATE_PATH + template_image_path)
+			)
+			# If we use ResourceLoader.load() it brings an CompressedTexture2D from a ctex file.
+			# so if we then use resource.duplicate() and try to save it with ResourceSaver.save()
+			# to a PNG file, it fails with ERR_FILE_UNRECOGNIZED.
+			image_path = storyquest_path + "/" + template_image_path.replacen("template_", "")
+			error = DirAccess.copy_absolute(TEMPLATE_PATH + template_image_path, image_path)
+			assert(error == OK)
+
+		# The following should be enough instead of a full scan, but it doesn't work:
+		# EditorInterface.get_resource_filesystem().update_file(storyquest_path + dialogue_path)
+		# EditorInterface.get_resource_filesystem().reimport_files(
+		# 	[storyquest_path + dialogue_path]
+		# )
+		EditorInterface.get_resource_filesystem().scan()
+		if EditorInterface.get_resource_filesystem().is_scanning():
+			await EditorInterface.get_resource_filesystem().resources_reimported
+
+		var dialogue_uid := ResourceUID.id_to_text(
+			ResourceLoader.get_resource_uid(storyquest_path + dialogue_path)
+		)
+
+		var image_uid: String
+		if image_path:
+			image_uid = ResourceUID.id_to_text(ResourceLoader.get_resource_uid(image_path))
+
+		var scene_contents = FileAccess.get_file_as_string(
+			TEMPLATE_PATH + template_info[i]["scene_path"]
+		)
+
+		scene_contents = scene_contents.replace(template_scene_uid, storyquest_uids[i])
+		scene_contents = scene_contents.replace(template_dialogue_uid, dialogue_uid)
```

Comment:
I'm interested that you're implementing this with search-and-replace on the raw `.tscn` file contents rather than manipulating the scene object and then saving it back – any particular reason?